frmts/ogcapi/gdalogcapidataset.cpp

```diff
@@ -1877,7 +1990,15 @@ bool OGCAPIDataset::InitWithTilesAPI(GDALOpenInfo *poOpenInfo,
             CPLGetConfigOption("GDAL_WMS_MAX_CONNECTIONS", "5")));
         const char *pszTileMatrix =
             CSLFetchNameValue(poOpenInfo->papszOpenOptions, "TILEMATRIX");
-        const int l_nBands = ((osRasterURL == osPNG_URL) ? 4 : 3);
+
+        std::unique_ptr<GDALDataset> dataset =
```

Comment:
opening a sample tile has some cost. It would be preferrable if this could be avoided in common cases like JPEG or PNG

Reply:
See comment above.